### PR TITLE
Finalize refactoring for profile plot2

### PR DESCRIPTION
**Pull request by agenius-mohammed-ali**

### CODE REVIEW

**Review by saranya-ag — CHANGES_REQUESTED** [review]

File: R/profilePlot.R [hunk 1]
```diff
@@ -112,115 +115,213 @@ profilePlot <- function(contrastDF,
...
                         footnoteColor = "black") {
-
+    ##### Asserts
+    assertthat::assert_that(!missing(contrastDF),
+                            !is.null(contrastDF),
+                            "data.frame" %in% class(contrastDF),
+                            nrow(contrastDF) > 0,
+                            msg = "contrastDF must be specified as dataframe with LogIntensity and LogRatio columns and optionally a p-value")
+    plotType <- tolower(plotType)
+    assertthat::assert_that(is.character(plotType),
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:142]: It is handled as converting plotType to lower will results in a 0 length character vector if plotType is **NULL** and won't be a NULL again.

File: R/profilePlot.R [hunk 2]
```diff
@@ -112,115 +115,213 @@ profilePlot <- function(contrastDF,
...
+    if (!is.null(lineFitType) &&
+        any(length(lineFitType) != 1,
+            !tolower(lineFitType) %in% c('glm', 'lm', 'loess', 'gam'))) {
+        warning("lineFitType must be one of 'glm', 'lm', 'loess', 'gam' or NULL to disable. Assigning default value 'loess'.")
+        lineFitType <- "loess"
+    }
+
+    if (!is.null(lineFitType) &&
+        any(is.null(lineFitColor),
+            !is.character(lineFitColor),
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:203]: done

File: R/profilePlot.R [hunk 3]
```diff
@@ -112,115 +115,213 @@ profilePlot <- function(contrastDF,
...
+            length(lineFitColor) != 1)) {
+        warning("lineFitColor must be a singular value of class character. Assigning default value 'goldenrod1'.")
+        lineFitColor <- "goldenrod1"
+    }
+
+    if (any(is.null(symbolSize),
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:209]: Why? if the user supplied 0 is numeric value users can use if they want, right?

File: R/profilePlot.R [hunk 4]
```diff
@@ -112,115 +115,213 @@ profilePlot <- function(contrastDF,
...
+    if (any(is.null(transparency),
+            !is.numeric(transparency),
+            length(transparency) != 1,
+            transparency <= 0,
+            transparency > 1)) {
+        warning("transparency must be a singular value of class numeric and must be between 0 and 1. Assigning default value '0.5'.")
+        transparency <- 0.5
+    }
+
+    if (!is.null(referenceLine) &&
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:245]: done

File: R/profilePlot.R [hunk 5]
```diff
@@ -112,115 +115,213 @@ profilePlot <- function(contrastDF,
...
+        !all(is.character(referenceLine), length(referenceLine) == 1)) {
+        warning("referenceLine must be a singular value of class character or 'NULL' to disable. Assigning default value 'darkgoldenrod1'.")
+        referenceLine <- "darkgoldenrod1"
     }
 
-    if (is.null(ylab)) {
-        ylab <- logRatioCol
+    if (any(is.null(refLineThickness),
+            !is.numeric(refLineThickness),
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:252]: Why? if the user supplied 0 is numeric value users can use if they want, right?

File: R/profilePlot.R [hunk 6]
```diff
@@ -112,115 +115,213 @@ profilePlot <- function(contrastDF,
...
+    if (!is.null(footnote) &&
+        !is.null(footnoteSize) &&
+        any(!is.numeric(footnoteSize),
+            length(footnoteSize) != 1)) {
+        warning("footnoteSize must be a singular value of class numeric. Assigning default value '3'.")
+        footnoteSize <- 3
+    }
+
+    if (!is.null(footnote) &&
+        !is.null(footnoteColor) &&
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:283]: done

File: R/profilePlot.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:209]: done

File: R/profilePlot.R (hunk 5 above)
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/profilePlot.R:252]: done

**Pull request closed without merging**